sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOrcScan.scala

```diff
@@ -227,23 +261,42 @@ object GpuOrcScan extends Arm {
   def canCast(from: TypeDescription, to: TypeDescription): Boolean = {
     import org.apache.orc.TypeDescription.Category._
     if (!to.getCategory.isPrimitive || !from.getCategory.isPrimitive) {
+      // List, Map, Struct, Union is not primitive
       // Don't convert from any to complex, or from complex to any.
       // Align with what CPU does.
       return false
     }
+    val toType = to.getCategory
     from.getCategory match {
       case BOOLEAN | BYTE | SHORT | INT | LONG =>
-        to.getCategory match {
-          case BOOLEAN | BYTE | SHORT | INT | LONG => true
+        toType match {
+          case BOOLEAN | BYTE | SHORT | INT | LONG | FLOAT | DOUBLE | STRING |
+               TIMESTAMP => true
+          // BINARY and DATE are not supported by design.
+          // The 'to' type (aka read schema) is from Spark, and VARCHAR and CHAR will
+          // be replaced by STRING. Meanwhile, cuDF doesn't support them as output
+          // types, and also replaces them with STRING.
+          // TIMESTAMP_INSTANT is not supported by cuDF.
           case _ => false
         }
-      case VARCHAR =>
-        to.getCategory == STRING
-      // TODO more types, tracked in https://github.com/NVIDIA/spark-rapids/issues/5895
-      case _ =>
-        false
+      case CHAR | VARCHAR =>
+        toType == STRING
+      case _ => false
     }
   }
+
+  /**
+   * Check whether if integerVec can be multiplied with k. If val * k / k == val, then there is no
+   * integer-overflow, we can make a multiplication. Otherwise, there is integer-overflow.
+   *
+   * @param integerVec a column vector with type of INT64 integers
+   * @param k          the number to be multiplied
+   */
+  def canMultiply(integerVec: ColumnView, k: Long): Boolean = {
+    assert(integerVec.getType() == DType.INT64)
+    val maxVal = integerVec.max().getLong
+    maxVal * k / k == maxVal
```

Comment:
If `k == 0` there would be a divide by zero error.
If `k == 1`, we can always return `true`.

Reply:
> If k == 0 there would be a divide by zero error.
Yep, we should return `k == 0 || maxval * k / k == maxval` here.

> the compiler could elide `* k / k`, or even simplify the whole expression to a constant `true`.

This will not happen, since `k` is not a constant value. The compiler can not optimize values that cannot be inferred at compile time.

Reply:
> This will not happen, since k is not a constant value. The compiler can not optimize values that cannot be inferred at compile time.

Test on c++ clang 14 Risc-v compiler with -O3 optimization:
https://godbolt.org/z/Pc3babMro

Reply:
In CPU code, `Math.multiplyExact(x, y)` will handle the overflow problems. So I did the similar thing.
```scala
val k = 1e6.toLong
Math.multiplyExact(col.max().getLong, k)
Math.multiplyExact(col.min().getLong, k)
```

If there is integer-overflow, then Math.multiplyExact will throw ArithmeticException, which is same as CPU code.